docs/getting-started/quick_start.md

```diff
@@ -60,31 +92,76 @@ It should be in iterable, yielding tuples, of the same length as the number of a
 
 <!--pytest-codeblocks:skip-->
 ```python
-inputset = [(2, 3), (0, 0), (1, 6), (7, 7), (7, 1)]
+max_value_included = 7
+inputset = [
+    (
+        np.random.randint(max_value_included + 1),
+        np.random.randint(max_value_included + 1),
+    )
+    for _ in range(10)
+]
 ```
 
+Here, our inputset is made of 10 pairs of integers, whose maximum value is `max_value_included`. Remark that the values which will later be used as inputs of the compiled function need to be in the same range.
+
 {% hint style="warning" %}
-All inputs in the inputset will be evaluated in the graph, which takes time. If you're experiencing long compilation times, consider providing a smaller inputset.
+All inputs in the inputset will be evaluated in the graph, which takes time. If you're experiencing long compilation times, consider providing a smaller inputset. At the same time, a too-short inputset means that the bit widths which are measured during the tracing may not be exact, and the consequence would be that the function correctness is not ensured.
 {% endhint %}
 
 ## Compiling the function
 
-You can use the `compile` method of a `Compiler` class with an inputset to perform the compilation and get the resulting circuit back:
+You can use the `compile` method of the `Compiler` class with an inputset to perform the compilation and get the resulting circuit back:
 
 <!--pytest-codeblocks:skip-->
 ```python
+print(f"Compiling...")
 circuit = compiler.compile(inputset)
 ```
 
+## Generate the keys
+
+You can use the `keygen` method of the `Circuit` class to generate the keys (public and private):
+
+<!--pytest-codeblocks:skip-->
+```python
+print(f"Generating keys...")
+circuit.keygen()
+```
+
 ## Performing homomorphic evaluation
 
-You can use the `encrypt_run_decrypt` method of a `Circuit` class to perform homomorphic evaluation:
+You can now iterate over different random samples, and use the `encrypt`, `run` and `decrypt` methods of the `Circuit` class to perform homomorphic evaluation:
 
 <!--pytest-codeblocks:skip-->
 ```python
-homomorphic_evaluation = circuit.encrypt_run_decrypt(4, 4)
+for _ in range(4):
+
+    example = (
+        np.random.randint(max_value_included + 1),
+        np.random.randint(max_value_included + 1),
+    )
+    encrypted_example = circuit.encrypt(*example)
+    encrypted_result = circuit.run(encrypted_example)
+    result = circuit.decrypt(encrypted_result)
 ```
 
-{% hint style="info" %}
-`circuit.encrypt_run_decrypt(*args)` is just a convenient way to do everything at once. It is implemented as `circuit.decrypt(circuit.run(circuit.encrypt(*args)))`.
-{% endhint %}
+Remark that there is a convenient method `encrypt_run_decrypt` to be used to simplify tests. The previous code snippet becomes:
```

Comment:
It's not just for `simplify tests` it can be used in many contexts. I'd say:
```
Concrete provides a convenient method `encrypt_run_decrypt` to do make all of those steps in one go. It's useful for development, demos, testing, etc. With this method the previous code snippet becomes:
```